test/unit/tool_util/test_required_files.py

```diff
@@ -0,0 +1,170 @@
+import os
+from pathlib import Path
+
+from .test_parsing import BaseLoaderTestCase
+
+SCRIPT_DIRECTORY = os.path.abspath(os.path.dirname(__file__))
+TEST_TOOL_DIRECTORIES = os.path.join(SCRIPT_DIRECTORY, "tool_directories")
+
+
+TOOL_REQUIRED_FILES_XML_1 = """
+<tool name="required_files_1" id="required_files_1" version="0.1.0">
+    <command>foo</command>
+    <required_files>
```

Comment:
I really like this, and i think we can use this in planemo as well for containerized jobs that need files that are symlinked in. Not sure how exactly that would look like but this would make it easier to find a solution that is not what we did now ... copying everything.